### PR TITLE
Fix the URL handling by using URL.Path instead of String() + various fix and features

### PR DESCRIPTION
**Pull request by jeremiejig**
Fix handling of URL with URL.Path
+ fix size comparison

+ possibility to use address:port in settings.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -619,7 +623,7 @@ func main() {
 		log.Println("server started")
 		for running {
 			time.Sleep(1 * time.Second)
-			for uint64(settings.MaxCacheSizeInMebibytes)*1024 < diskUsed {
+			for uint64(settings.MaxCacheSizeInMebibytes) < diskUsed {
```
Comment (columna1) [inline main.go:626]: diskUsed is in bytes I'm pretty sure so I'm not sure this will work as intended

File: main.go [hunk 2]
```diff
@@ -470,7 +474,7 @@ func readSettingsFile() bool {
 			log.Println("Client secret need to be a 52 char alphanumeric string")
 			return false
 		}
-		if settings.MaxCacheSizeInMebibytes < 10240 {
+		if settings.MaxCacheSizeInMebibytes < 10240*1024*1024 {
```
Comment (columna1) [inline main.go:477]: is this not 10485760 gibibits?

**Review by columna1 — COMMENTED** [review]
I think you made a mistake, the setting is mebibytes not bytes

File: main.go (hunk 1 above)
Comment (jeremiejig) [reply main.go:626]: Well yes and with your new loading of settings MaxCacheSizeInMebibytes is now in bytes too.

File: main.go (hunk 2 above)
Comment (jeremiejig) [reply main.go:477]: Well Kind of got lazy there, and I think the compiler might optimise this constant,

If it does not you only do that at the launch of the program so...

Comment (columna1) [reply main.go:477]: oh duhh, yeah I forgot about that, thanks.

**Comment by columna1** [comment]
Thanks a bunch 👍

**Pull request merged**